src/java.desktop/share/classes/javax/swing/ToolTipManager.java

```diff
@@ -301,8 +307,12 @@ void showTipWindow() {
             location.x -= size.width;
         }
             } else {
-                location = new Point(screenLocation.x + mouseEvent.getX(),
-                        screenLocation.y + mouseEvent.getY() + 20);
+                if (mouseEvent != null) {
```

Comment:
this may not be in the scope of changes but is it worth spending some time for code formatting ? Looks very messy.

Reply:
may be as a cleanup later on, the whole file formatting is off at many places..